planner/core/logical_plan_builder.go

```diff
@@ -3797,9 +3799,16 @@ func (b *PlanBuilder) buildSelect(ctx context.Context, sel *ast.SelectStmt) (p L
 	}
 
 	// Determines whether to use the Merge hint in a CTE query.
```

Comment:
Please descript the rule here.

Reply:
Rules for determining Merge hint have been added